src/stcal/outlier_detection/median.py

```diff
@@ -0,0 +1,363 @@
+                 ) -> None:
+        """
+        Set up temporary files to perform operations on a stack of 2-D input
+        arrays along the stacking axis (e.g., a time axis) without
+        holding all of them in memory. Currently the only supported operation
+        is the median.
+
+        Parameters
+        ----------
+        shape
+            The shape of the entire input, (n_images, imrows, imcols).
+
+        dtype
+            The data type of the input data.
+
+        tempdir
+            The parent directory in which to create the temporary directory,
+            which itself holds all the DiskAppendableArray tempfiles.
+            Default is the current working directory.
+
+        buffer_size
+            The buffer size, units of bytes.
+            Default is the size of one input image.
+        """
+        if len(shape) != 3:
+            msg = f"Invalid input shape {shape}; "
+            msg += "only three-dimensional data are supported."
+            raise ValueError(msg)
+        self._expected_nframes = shape[0]
+        self.frame_shape = shape[1:]
+        self.dtype = np.dtype(dtype)
+        self.itemsize = self.dtype.itemsize
+        self._temp_dir = tempfile.TemporaryDirectory(dir=tempdir)
+        self._temp_path = Path(self._temp_dir.name)
+
+        # figure out number of sections and rows per section that are needed
+        self.nsections, self.section_nrows = \
+            self._get_buffer_indices(buffer_size=buffer_size)
+        self.slice_shape = (self.section_nrows, shape[2])
+        self._n_adds = 0
+
+        # instantiate a temporary DiskAppendableArray for each section
+        self._temp_arrays = self._temparray_setup(dtype)
+
+    def _get_buffer_indices(self,
+                            buffer_size: int = 0
+                            ) -> tuple[int, int]:
+        """
+        Determine the number of sections and rows per section needed to
+        divide the input data into sections that fit within the specified
+        buffer size.
+
+        Parameters
+        ----------
+        buffer_size
+            The buffer size for the median computation, units of bytes.
+
+        Returns
+        -------
+        nsections
+            The number of sections to divide the input data into.
+
+        section_nrows
+            The number of rows in each section (except the last one).
+        """
+        imrows, imcols = self.frame_shape
+        if buffer_size == 0:
+            buffer_size = imrows * imcols * self.itemsize
+        per_model_buffer_size = buffer_size / self._expected_nframes
+        min_buffer_size = imcols * self.itemsize
+        section_nrows = \
+            min(imrows, int(per_model_buffer_size // min_buffer_size))
+
+        if section_nrows <= 0:
+            buffer_size = min_buffer_size * self._expected_nframes
+            msg = "Buffer size is too small to hold a single row. "
+            msg += f"Increasing buffer size to {buffer_size / _ONE_MB} MB"
```

Comment:
```suggestion
            msg = (
                "Buffer size is too small to hold a single row. "
                f"Increasing buffer size to {buffer_size / _ONE_MB} MB"
            )
```
This is likely a nit pick but for multi-line strings this should pass the ruff rules and avoid the need to add strings (which results in re-allocations and copies). This comment applies to several places in this PR.